pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -100,8 +100,8 @@ type informationForClaim struct {
 	allocation *resourceapi.AllocationResult
 }
 
-// dynamicResources is a plugin that ensures that ResourceClaims are allocated.
-type dynamicResources struct {
+// DynamicResources is a plugin that ensures that ResourceClaims are allocated.
+type DynamicResources struct {
```

Comment:
Not meant to, but some things are done that way anyway:
- [Cluster Autoscaler](https://github.com/kubernetes/autoscaler/blob/a6a77b3ec3f3adbf6bbde43907f79c834fe6045f/cluster-autoscaler/go.mod#L57)
- any custom scheduler which uses scheduler plugins like those from https://github.com/kubernetes-sigs/scheduler-plugins

Reply:
just saying, reading the PR description it felt to me people may have some expectations that I think is important to clarify ... but looking at thoe links  I wonder if the scheduler shouldl work on moving that path to staging 🤔

Reply:
Hi, why is that? There are lots of plugins that can be consumed externally, volcano also introduced some k8s plugins in this way before. I discussed it with @pohly, DRA plugin wasn't mature enough so it can't be consumed externally, but after the volcano verification was completed, he felt that it would be ok to open DRA.

Reply:
Hi @aojea, thanks for your review. Not mean to directly open this PR, but I have discussed it with @pohly on slack,  DRA plugin wasn't mature enough so it can't be consumed externally, but after the volcano verification was completed, he felt that it would be ok to open DRA. I saw that lots of other scheduler plugins can also be consumed externally, volcano introduces them the same way. In fact, as long as volcano can access the DRA plugin externally, it doesn’t matter which dependent package the DRA plugin is in.

Reply:
> sig-scheduler should work on moving that path to staging

I agree that this would be cleaner.

I wasn't involved in discussions around it, but I suppose it's not getting done because it would be a major change: it's not sufficient to just move plugins or the framework, for the "custom scheduler" use case *everything* related to the scheduler would have to be under staging, similar to the apiserver. That's a lot of work for something that mostly just works the way it is...

Reply:
yeah, I was just curious about it, if people is happy with existing workflow and it works for everybody then it is ok

Reply:
cc @alculquicondor wdyt is the preferred pattern here? It seems that the changes proposed by @JesseStutler in this PR are the more common pattern for plugins, but strictly speaking abstracting behind a `framework.Plugin` seems cleaner (the way the DRA plugin currently looks, which this PR is advocating we change).

Reply:
According to my experience, most custom schedulers are vendoring default-scheduler for particular plugin's  implementation wthin their own framework. So things would get complicated and vague if we move things to staging - where we only place generic APIs for now - it'd be like our goal is to build sth. generic for all scheduler-like components, and users would treat them as backwards-compatible API which would be a wrong assumption.

Current workflow works better IMO - for plugin itself, it'd ok to public them as a contract of SDK level of implementation instead of API. And consumers are responsible to upgrade the SDK along with k8s version upgrade.